Makefile

```diff
@@ -17,7 +17,10 @@ create_database: .env
 	docker compose run server create_db
 
 clean:
-	docker compose down && docker compose rm
+	docker compose down && docker compose rm && \
+	docker image rm --force \
+		cypress-server:latest cypress-worker:latest cypress-scheduler:latest \
+		redash-server:latest redash-worker:latest redash-scheduler:latest
```

Comment:
Does this command make `docker compose rm` redundant?

Reply:
Quite possibly.  I just tacked my bit onto the end without really looking at it.

I'll do some mucking around tomorrow and see if the `docker compose rm` is still useful. :smile:

Reply:
Naming the images to remove makes sense to me, I do not think there is ever a reason to remove pgautoupgrade/pgautoupgrade, redis, maildev/maildev

Reply:
Yeah, those ones are pretty static.  Only removing the ones we rebuild on each build-test iteration seems the most sensible approach.

Reply:
In testing just now, the `docker compose rm` line seems to still be **somewhat** useful, as it's removing **some** (but not all) of the previously stopped `redash-*` containers.

`docker container ls --all` does still show containers from previous build runs though, even after the `docker image rm --force cypress-server:latest [etc]` command is run.

```
$ docker container ls --all
CONTAINER ID   IMAGE          COMMAND                  CREATED         STATUS                     PORTS     NAMES
85c48343ce7f   566aa01cf283   "/app/bin/docker-ent…"   3 minutes ago   Exited (0) 3 minutes ago             redash-server-run-19786059e708
c87a97ffea6f   566aa01cf283   "/app/bin/docker-ent…"   7 minutes ago   Exited (0) 6 minutes ago             redash-server-run-664d09667191
c80e15f03d3c   2039a2dabfdb   "/usr/local/bin/dock…"   2 weeks ago     Exited (0) 2 weeks ago               pgauto
```

Note the bottom line there is a container from when I was building pgautoupgrade containers a few weeks ago. :wink:

The only thing that seems to truly clear out all of the old stuff is the `docker [type] prune -f` family.  ie:

```
$ docker container prune -f
Deleted Containers:
85c48343ce7f2ecaf6c858cf1034dcd1a5c83d01e53c12b5306ace1e36b516d4
c87a97ffea6fe8f4d93333ea4447f97e7a9ed7954bdbfe38cea8472dbe9d2df1
c80e15f03d3ca88d27858d1c05fe18ea49ad5e173466533ff434bf94ba6d5174

Total reclaimed space: 0B
$ docker container ls --all
CONTAINER ID   IMAGE     COMMAND   CREATED   STATUS    PORTS     NAMES
$
```

Perhaps we should be adding `docker container prune -f`, `docker image prune -f`, and `docker volume prune -f` to the `make clean` target as well?

That will *really* remove any left over development images, containers, and volumes.

Reply:
It's strange how difficult it is to actually stop and remove all docker images. This seems to work well

```
docker compose rm --stop --force
docker container prune -f
docker image prune -f
docker volume prune -f
```

Then run `docker image rm ` as you suggested in this PR

Reply:
Did a bunch more testing and updated the PR with a bunch more commands, and added a new `make clean-all` target to even get rid of the alpine/postgres/old-redash images too. :smile: